frontend/src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/LegacyFeatureToggleSwitch.tsx

```diff
@@ -0,0 +1,257 @@
+    useFeatureStrategyProdGuard,
+} from '../../../../feature/FeatureStrategy/FeatureStrategyProdGuard/FeatureStrategyProdGuard';
+
+const StyledBoxContainer = styled(Box)<{ 'data-testid': string }>(() => ({
+    mx: 'auto',
+    ...flexRow,
+}));
+
+interface IFeatureToggleSwitchProps {
+    featureId: string;
+    environmentName: string;
+    projectId: string;
+    value: boolean;
+    onError?: () => void;
+    onToggle?: (
+        projectId: string,
+        feature: string,
+        env: string,
+        state: boolean,
+    ) => void;
+}
+
+/**
+ * @deprecated remove when flag `featureSwitchRefactor` is removed
+ */
+export const FeatureToggleSwitch: VFC<IFeatureToggleSwitchProps> = ({
+    projectId,
+    featureId,
+    environmentName,
+    value,
+    onToggle,
+    onError,
+}) => {
+    const { loading, toggleFeatureEnvironmentOn, toggleFeatureEnvironmentOff } =
+        useFeatureApi();
+    const { setToastData, setToastApiError } = useToast();
+    const { isChangeRequestConfigured } = useChangeRequestsEnabled(projectId);
+    const {
+        onChangeRequestToggle,
+        onChangeRequestToggleClose,
+        onChangeRequestToggleConfirm,
+        changeRequestDialogDetails,
+    } = useChangeRequestToggle(projectId);
+    const [isChecked, setIsChecked, rollbackIsChecked] =
+        useOptimisticUpdate<boolean>(value);
+
+    const [showEnabledDialog, setShowEnabledDialog] = useState(false);
+    const { feature } = useFeature(projectId, featureId);
+    const enableProdGuard = useFeatureStrategyProdGuard(
+        feature,
+        environmentName,
+    );
+    const [showProdGuard, setShowProdGuard] = useState(false);
+
+    const disabledStrategiesCount =
+        feature?.environments
+            .find((env) => env.name === environmentName)
+            ?.strategies.filter((strategy) => strategy.disabled).length ?? 0;
+
+    const handleToggleEnvironmentOn = async (
+        shouldActivateDisabled = false,
+    ) => {
+        try {
+            setIsChecked(!isChecked);
+            await toggleFeatureEnvironmentOn(
+                projectId,
+                feature.name,
+                environmentName,
+                shouldActivateDisabled,
+            );
+            setToastData({
+                type: 'success',
+                title: `Available in ${environmentName}`,
+                text: `${feature.name} is now available in ${environmentName} based on its defined strategies.`,
+            });
+            onToggle?.(projectId, feature.name, environmentName, !isChecked);
+        } catch (error: unknown) {
+            if (
+                error instanceof Error &&
+                error.message === ENVIRONMENT_STRATEGY_ERROR
+            ) {
+                onError?.();
+            } else {
+                setToastApiError(formatUnknownError(error));
+            }
+            rollbackIsChecked();
+        }
+    };
+
+    const handleToggleEnvironmentOff = async () => {
```

Comment:
Should follow the ADR of `onAction` for handlers

Reply:
`LegacyFeatureToggleSwitch.tsx`, not new code